src/components/elements/tab/tab.tsx

```diff
@@ -0,0 +1,112 @@
+import { Tab as DSTab } from "@lifesg/react-design-system/tab";
+import isEmpty from "lodash/isEmpty";
+import isObject from "lodash/isObject";
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { TestHelper } from "../../../utils";
+import { useValidationConfig } from "../../../utils/hooks";
+import { TFrontendEngineFieldSchema } from "../../frontend-engine";
+import { IGenericElementProps } from "../types";
+import { Wrapper } from "../wrapper";
+import { ITabSchema } from "./types";
+
+export const Tab = (props: IGenericElementProps<ITabSchema>) => {
+	// =========================================================================
+	// CONST, STATE, REF
+	// =========================================================================
+	const {
+		id,
+		schema: { currentActiveTabId, children },
+	} = props;
+	const [currentTabIndex, setCurrentTabIndex] = useState(getCurrentTabIndex());
+	const { removeFieldValidationConfig } = useValidationConfig();
+	const { unregister } = useFormContext();
+
+	// =========================================================================
+	// EFFECTS
+	// =========================================================================
+	useEffect(() => {
+		setCurrentTabIndex(getCurrentTabIndex());
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [currentActiveTabId]);
+
+	useEffect(() => {
+		Object.values(children).forEach((childSchema, index) => {
+			if (index === currentTabIndex) {
+				return;
+			}
+			const idsToDelete = listAllChildIds(childSchema);
+			idsToDelete.forEach((idToDelete) => {
+				removeFieldValidationConfig(idToDelete);
+				unregister(idToDelete);
+			});
+		});
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [currentTabIndex, children]);
```

Comment:
Found that this happens even when the tabs are uncontrolled. I suspect it's because `Wrapper` initialises the child component to null before loading the schema, as this problem goes away when changing `useDeepCompareEffect` to `useLayoutEffect`